docs/gt_frontend/language_reference.rst

```diff
@@ -0,0 +1,332 @@
+==========================================
+Language Reference
+==========================================
+
+Abstract
+========
+
+This document contains a draft of the DSL syntax for GTScript on meshes.
+
+**Design Goals**
+
+1. Concise, compact and readable syntax
+2. General enough to allow generation of efficient code for meshes with unstructured, partially structured, structured meshes
+3. Easy to translate into GTScriptAST
+
+Motivation and Scope
+====================
+
+For global weather models (e.g. `IFS-FVM <https://refubium.fu-berlin.de/bitstream/handle/fub188/25122/K%C3%BChnlein_FVM_2019.pdf?sequence=1>`__, ICON) the ability to handle non-regular grids is required.
+
+Concepts
```

Comment:
I would add a TODO here mentioning that this part is WIP and it should be moved to the concepts repo soon

Reply:
Agreed